### ISSUE TITLE
[Issue]: Transcoding speed steadily declines - AMD VAAPI 5650G APU

### ISSUE THREAD

**Issue by iwch** [jellyfin/jellyfin#9691]
### Please describe your bug

Playing almost any media which requires transcoding on my Jellyfin instance via the web client starts fine, but the transcoding rate drops steadily after playback starts. As you can see in the log I attached, the transcode rate starts just above 1.5x, and the playback is initially smooth and trouble free. However, for whatever reason the transcode rate steadily drops, eventually going below even 0.9x, at which point the playback becomes a stuttery mess. The problem occurs even for resolutions which the GPU should be capable of easily handling (1080p 30fps), though I attached logs from the most difficult case (4K HDR10 H.265 -> 4K SDR H.264 120Mbps). I encountered the issue in both Chrome and Firefox.

Everything works flawlessly when I direct play the same files via DLNA using VLC.

The GPU I use is a Renoir APU (5650G), using the 5.19 kernel. I installed ROCm OpenCL in the docker container to do the tonemapping, though the issue still exists if I entirely disable tonemapping or use an SDR video.

Additionally, running the jellyfin-ffmpeg command run by Jellyfin itself directly on the command line, I noticed that the speed dip occurs always right after jellyfin-ffmpeg opens a .ts file in `/config/transcodes/<jumble of letters and numbers>.ts`. The first few of these files are written very quickly. but after a while each file takes a long time to open and start writing to. 

I'd greatly appreciate help to pinpoint the issue behind the reason why the transcoding rate slows down so much after starting reasonably quick.

In case it's useful, I see the following error messages repeated a few times in the kernel logs:

```
[  563.938356] amdgpu: init_user_pages: Failed to get user pages: -1
[  563.938567] amdgpu: No more SDMA queue to allocate
[  563.938569] amdgpu: Pasid 0x8005 DQM create queue type 1 failed. ret -12
[  563.938991] amdgpu: No more SDMA queue to allocate
[  563.938993] amdgpu: Pasid 0x8005 DQM create queue type 1 failed. ret -12
```

### Jellyfin Version

10.8.z

### Environment

```markdown
- OS: Ubuntu 22.04, using the official docker image
- Linux Kernel: 5.19
- Virtualization: None
- Clients: Web client
- Browser: Latest Firefox, Chrome
- FFmpeg Version: Version in the docker image
- Playback Method: Transcode H.265->H.264
- Hardware Acceleration: VAAPI
- GPU Model: AMD 5650G
- Memory: 2x8GB DDR4 2666MHz ECC
- Plugins: -
- Reverse Proxy: -
- Base URL: -
- Networking: Local LAN
- Storage: Media on ZFS RAIDZ2 pool with 5x 8TB HDDs, OS installed on SATA SSD.
```

### FFmpeg logs

```shell
https://pastebin.com/tEdHt4Ac
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nyanmisaka** [comment]
It’s time to blame AMD:

https://gitlab.freedesktop.org/mesa/mesa/-/issues/8313

**Comment by iwch** [comment]
@nyanmisaka Thanks for the response.

I fixed a couple of issues with my system that were not related to the GPU which helped things. Specifically, I figured out that the SSD I was writing the transcodes into was choking hard after a while (it's an old model). Changing the transcode path to a directory on my ZFS pool helped massively.

I'm still seeing the slowdown over time - 4K HEVC HDR movies start at ~1.55x and slow down to ~1.33x - but I can now stream an entire transcoded movie with only a handful hitches. I think the issue with mesa you linked is certainly there (`watch -n 0.2 "rocm-smi -c"` does show the GPU constantly jumps between 1900MHz and 400MHz) but it looks like there isn't much that Jellyfin developers can do about that, so feel free to close this issue.
